pkg/shared/utils/utils.go

```diff
@@ -57,6 +57,14 @@ func PointerTo[T any](value T) *T {
 	return &value
 }
 
+func PointerToStringOrNil(s string) *string {
+	if s == "" {
+		return nil
+	}
+
+	return &s
+}
```

Comment:
I wonder if we can make a generic here which checks for the zero type. Something like:

```
func OmitEmpty[T comparable](t T) *T {
	var empty T
	if t == empty {
		return nil
	}
	return &t
}
```
https://go.dev/play/p/UW1rdYWe_V2

This works for all comparable types, but doesn't work for things like []string unfortunately.